base/setup/reactos/drivepage.c

```diff
@@ -95,27 +97,271 @@ MoreOptDlgProc(HWND hwndDlg,
     return FALSE;
 }
 
-static INT_PTR CALLBACK
-PartitionDlgProc(HWND hwndDlg,
-                 UINT uMsg,
-                 WPARAM wParam,
-                 LPARAM lParam)
+
+/**
+ * @brief   Data structure stored for each partition entry in the TreeList.
+ **/
+typedef struct _PARTITEM
 {
+    PPARTENTRY PartEntry; //< Disk region this structure is associated to.
+    PVOLENTRY Volume;     //< Associated file system volume if any, or NULL.
+    PVOL_CREATE_INFO VolCreate; //< Volume creation parameters.
+} PARTITEM, *PPARTITEM;
+
+/**
+ * @brief   Dialog context structure used by PartitionDlgProc().
+ **/
+typedef struct _PARTCREATE_CTX
+{
+    PARTITEM PartItem; // A copy of the info stored in the TreeList
+    ULONG MaxSizeMB;
+    ULONG PartSizeMB;
+    BOOLEAN MBRExtPart;
+} PARTCREATE_CTX, *PPARTCREATE_CTX;
+
+static INT_PTR
+CALLBACK
+PartitionDlgProc(
+    _In_ HWND hDlg,
+    _In_ UINT uMsg,
+    _In_ WPARAM wParam,
+    _In_ LPARAM lParam)
+{
+    PPARTCREATE_CTX PartCreateCtx;
+
+    /* Retrieve dialog context pointer */
+    PartCreateCtx = (PPARTCREATE_CTX)GetWindowLongPtrW(hDlg, GWLP_USERDATA);
+
     switch (uMsg)
     {
         case WM_INITDIALOG:
+        {
+            PPARTENTRY PartEntry;
+            PDISKENTRY DiskEntry;
+            ULONG MaxSizeMB;
+            ULONG Index = 0;
+            INT iItem;
+            PCWSTR FileSystemName;
+            PCWSTR DefaultFs;
+
+            /* Save dialog context pointer */
+            PartCreateCtx = (PPARTCREATE_CTX)lParam;
+            SetWindowLongPtrW(hDlg, GWLP_USERDATA, (LONG_PTR)PartCreateCtx);
+
+            /* Retrieve the selected partition */
+            PartEntry = PartCreateCtx->PartItem.PartEntry;
+            DiskEntry = PartEntry->DiskEntry;
+
+            /* Set the spinner to the maximum size in MB the partition can have */
+            MaxSizeMB = PartCreateCtx->MaxSizeMB;
+            SendDlgItemMessageW(hDlg, IDC_UPDOWN_PARTSIZE, UDM_SETRANGE32, (WPARAM)1, (LPARAM)MaxSizeMB);
+            SendDlgItemMessageW(hDlg, IDC_UPDOWN_PARTSIZE, UDM_SETPOS32, 0, (LPARAM)MaxSizeMB);
+            // SetDlgItemInt(hDlg, IDC_EDIT_PARTSIZE, MaxSizeMB, FALSE);
```

Comment:
```suggestion
```
prepared for the future or forgotten to remove?

Reply:
I still have to investigate a problem of setting correctly the number with the spinner control.

Reply:
I've removed that commented line: the text is automatically set into the edit field when setting the updown control value to the intended value (the `SendDlgItemMessageW(hDlg, IDC_UPDOWN_PARTSIZE, UDM_SETPOS32, 0, (LPARAM)MaxSizeMB);` line just above does the job).